### PR TITLE
SLE-11-SP3 backports

### PR DESCRIPTION
**Pull request by maltek**

### CODE REVIEW

**Comment by mgerstner** [comment]
Could you add a `Makefile` on this branch to match the situation on other branches somewhat?

File: chkstat.c [hunk 1]
```diff
@@ -492,19 +598,21 @@ safe_open(char *path, struct stat *stb, uid_t target_uid, bool *traversed_insecu
...
+      if (!S_ISLNK(stb->st_mode) && (stb->st_mode & S_IWOTH) && !is_final_path_element)
         *traversed_insecure = true;
       // if parent directory is not owned by root, the file owner must match the owner of parent
       if (stb->st_uid && stb->st_uid != target_uid && stb->st_uid != euid)
         {
-          if (p)
-            goto fail_insecure_path;
+          if (is_final_path_element)
+            {
+              //fprintf(stderr, "%s: has unexpected owner. refusing to correct due to unknown integrity.\n", path+rootl);
```
Comment (mgerstner) [inline chkstat.c]: Could you also add the reasoning as in SLE-15-SP1 here? There you have:

`// do not backport behavior change`

**Comment by mgerstner** [comment]
Running the testsuite (that you also merged into the new branch) the following test cases currently fail:

1) TestCapabilities: the codebase in SLE-11 (and SLE-11 as such?) doesn't support capabilities
2) TestUnexpectedPathOwner: the behaviour change is not backported, like in SLE-15-SP1
3) TestPrivsForSpecialFiles: FIFO special file target is not rejected. Probably like 2) ?
4) TestPrivsOnInsecurePath: chkstat can't parse the `+capabilities` line and thus still applies the 
      previously encountered base line for the file.

Could you disable/adjust the test cases so that they run correctly on the branch? Also adding the `make test` target in the `Makefile` mentioned before would be nice.

**Comment by mgerstner** [comment]
> Could you disable/adjust the test cases so that they run correctly on the branch? Also adding the make test target in the Makefile mentioned before would be nice.

Sorry I was mistaken. I thought you commited the regtests, but they were only local remnants in my working copy. So if you agree to the test results I mentioned before then there's nothing else to do regarding this.

When adding a `Makefile` then we can also switch the spec file in this codestream to actually use `make` instead of hard coding gcc invocations in there ...

**Comment by maltek** [comment]
> TestPrivsForSpecialFiles: FIFO special file target is not rejected. Probably like 2) ?

same as 1: this tests something that chkstat doesn't assign a capability for a FIFO, and there's no capability support in the code stream.

> Sorry I was mistaken. I thought you commited the regtests, but they were only local remnants in my working copy.

Indeed, I was too lazy to backport it to all code streams. Since running it requires a recent kernel any way, IMHO it's easier to just maintain the tests in master / new forks and run them against the older code streams from there.

> So if you agree to the test results I mentioned before then there's nothing else to do regarding this.

Yes, test suite results look good to me.

File: chkstat.c (hunk 1 above)
Comment (maltek) [reply chkstat.c]: Done. That's a remnant of manually editing patch files in different code streams. (I think one also uses "don't" instead of "do not".)

Comment (mgerstner) [reply chkstat.c]: I think there's no strict rule to use "don't", actually I've been told in formal texts "do not" is to be preferred.

**Comment by mgerstner** [comment]
Now when building with the new `Makefile` it reports a warning:

```
gcc -W -Wall -g -O2    chkstat.c   -o chkstat
chkstat.c: In function ‘main’:
chkstat.c:731:15: warning: unused variable ‘str’ [-Wunused-variable]
  731 |   char *opt, *str;
      |               ^~~
```
Nothing serious but maybe you/we can fix it after I merge the PR.

**Pull request merged**